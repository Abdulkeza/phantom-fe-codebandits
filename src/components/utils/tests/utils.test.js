import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import * as redux from 'react-redux';
import Roles from '../../../pages/dashboard/RolesPage.js';
import CheckRole from '../CheckRoles.js';


let spyOnUseSelector;

const mockUseLocationValue = {
  pathname: '/testroute',
  search: '',
  hash: '',
  state: null
};
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: jest.fn().mockImplementation(() => {
    return mockUseLocationValue;
  })
}));
describe('Check roles', () => {
  spyOnUseSelector = jest.spyOn(redux, 'useSelector');
  it('Return the element for the admin', () => {
    spyOnUseSelector.mockReturnValue({
      authenticated: true,
      user: {
        id: 2,
        firstName: 'Fabrice',
        email: 'admin@me.com',
        roles: {
          name: 'admin',
          permissions: [
            {
              driver: ['get', 'create', 'update', 'delete'],
              routes: ['get', 'create', 'update', 'delete'],
              buses: ['get', 'create', 'update', 'delete'],
              operator: ['get', 'create', 'update', 'delete'],
              company: ['get', 'create', 'update', 'delete']
            }
          ]
        }
      }
    });
    const elem = shallow(
      <CheckRole role={['admin']} children={<Roles></Roles>} />
    );
    expect(elem).toMatchSnapshot();
  });
  it('Return the element for the operator', () => {
    spyOnUseSelector.mockReturnValue({
      authenticated: true,
      user: {
        id: 2,
        firstName: 'Fabrice',
        email: 'admin@me.com',
        roles: {
          name: 'operator',
          permissions: [
            {
              driver: ['get', 'create', 'update', 'delete'],
              routes: ['get', 'create', 'update', 'delete'],
              buses: ['get', 'create', 'update', 'delete'],
              operator: ['get', 'create', 'update', 'delete'],
              company: ['get', 'create', 'update', 'delete']
            }
          ]
        }
      }
    });
    const elem = shallow(
      <CheckRole role={['operator']} children={<Roles></Roles>} />
    );
    expect(elem).toMatchSnapshot();
  });
  it('Return empty for for the operator', () => {
    spyOnUseSelector.mockReturnValue({
      authenticated: true,
      user: {
        id: 2,
        firstName: 'Fabrice',
        email: 'admin@me.com',
        roles: {
          name: 'operator',
          permissions: [
            {
              driver: ['get', 'create', 'update', 'delete'],
              routes: ['get', 'create', 'update', 'delete'],
              buses: ['get', 'create', 'update', 'delete'],
              operator: ['get', 'create', 'update', 'delete'],
              company: ['get', 'create', 'update', 'delete']
            }
          ]
        }
      }
    });
    const elem = shallow(
      <CheckRole type={'page'} role={['admin']} children={<Roles></Roles>} />
    );
    expect(elem).toMatchSnapshot();
  });
  it('Return empty for for the operator', () => {
    spyOnUseSelector.mockReturnValue({
      authenticated: true,
      user: {
        id: 2,
        firstName: 'Fabrice',
        email: 'admin@me.com',
        roles: {
          name: 'operator',
          permissions: [
            {
              driver: ['get', 'create', 'update', 'delete'],
              routes: ['get', 'create', 'update', 'delete'],
              buses: ['get', 'create', 'update', 'delete'],
              operator: ['get', 'create', 'update', 'delete'],
              company: ['get', 'create', 'update', 'delete']
            }
          ]
        }
      }
    });
    const elem = shallow(
      <CheckRole role={['admin']} children={<Roles></Roles>} />
    );
    expect(elem).toMatchSnapshot();
  });
});
