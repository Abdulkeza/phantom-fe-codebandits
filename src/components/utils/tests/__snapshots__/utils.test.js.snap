// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check roles Return empty for for the operator 1`] = `
<div
  className="flex flex-col items-center justify-center"
>
  <h3
    className="font-rale font-bold"
  >
    Don't have the permissions to view the page.
  </h3>
  <Link
    className="bg-primary rounded-sm font-bold"
    to="/login"
  >
    Login with a different account
  </Link>
</div>
`;

exports[`Check roles Return empty for for the operator 2`] = `""`;

exports[`Check roles Return the element for the admin 1`] = `<Roles />`;

exports[`Check roles Return the element for the operator 1`] = `<Roles />`;
